snf-cyclades-app/synnefo/userdata/models.py

```diff
@@ -57,9 +58,23 @@ class PublicKeyPair(ProfileModel):
     content = models.TextField(validators=[
         MaxLengthValidator(SSH_KEY_MAX_CONTENT_LENGTH)])
     fingerprint = models.CharField(max_length=100, null=False, blank=True)
+    created_at = models.DateTimeField(auto_now_add=True,
+                                      default=datetime.datetime.\
+                                      fromtimestamp(0))
+    updated_at = models.DateTimeField(auto_now=True, null=True)
+    # These fields should always be null and
+    # false respectively
+    deleted_at = models.DateTimeField(null=True, default=None)
+    deleted = models.BooleanField(null=False, default=False)
+    # Type field should take values between ssh and x509
+    # Django does not currently support
+    # SQL ENUM type so we wiil just use
+    # a plain CharField
+    type = models.CharField(max_length=10, null=False, default='ssh')
```

Comment:
My original idea was to use something that maps to postresql's ENUM type but Django doesn't have something built in. Using choices will only serve as a validation mechanism which is great. To further improve this I suggest creating a custom field that uses ENUM.
More can be found [here](https://docs.djangoproject.com/en/dev/howto/custom-model-fields/#howto-custom-model-fields)